### PR TITLE
Make circular buffer op configurable via flatbuffers.

### PR DESCRIPTION
**Pull request by njeffrie**
Make circular buffer op configurable via flexbuffer arguments.

Fix circular_buffer test on bluepill.

Base change: http://cl/360707883

http://b/143286954
http://b/150001379

### CODE REVIEW

File: tensorflow/lite/micro/tools/make/Makefile [hunk 1]
```diff
@@ -313,6 +313,7 @@ tensorflow/lite/micro/kernels/arg_min_max.cc \
 tensorflow/lite/micro/kernels/batch_to_space_nd.cc \
 tensorflow/lite/micro/kernels/cast.cc \
 tensorflow/lite/micro/kernels/ceil.cc \
+tensorflow/lite/micro/kernels/circular_buffer_flexbuffers_generated_data.cc \
```
Comment (njeffrie) [inline tensorflow/lite/micro/tools/make/Makefile:316]: It feels like this shouldn't be in the KERNEL_SRCS, but that's where the other flexbuffers generated data lives, and it does not build if it's only part of the tests.

File: tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc [hunk 2]
```diff
@@ -66,8 +66,7 @@ EXCLUDED_TESTS := \
   tensorflow/lite/micro/micro_interpreter_test.cc \
   tensorflow/lite/micro/micro_allocator_test.cc \
   tensorflow/lite/micro/memory_helpers_test.cc \
-  tensorflow/lite/micro/memory_arena_threshold_test.cc \
-  tensorflow/lite/micro/kernels/circular_buffer_test.cc
+  tensorflow/lite/micro/memory_arena_threshold_test.cc
```
Comment (advaitjain) [inline tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc:69]: let's also enable for the stm32f4 target

**Review by advaitjain — APPROVED** [review]
Only small nits.

Let's link to these bugs from the PR description:
http://b/143286954
http://b/150001379

And if you're up for it, let's also do a followup PR that cleans up the names etc. for the detection_postprocess_test's flexbuffer data generation.

File: tensorflow/lite/micro/tools/make/Makefile (hunk 1 above)
Comment (advaitjain) [reply tensorflow/lite/micro/tools/make/Makefile:316]: I think adding an explicit test in the Makefile will help here.

Something like:
https://github.com/tensorflow/tensorflow/blob/6a2d982fa971ef803df4adf5dda449431b76ce3e/tensorflow/lite/micro/tools/make/Makefile#L679-L691

File: tensorflow/lite/micro/kernels/BUILD [hunk 3]
```diff
@@ -449,12 +449,23 @@ cc_test(
...
 
+cc_library(
+    name = "circular_buffer_flexbuffers_generated_data",
+    srcs = [
+        "circular_buffer_flexbuffers_generated_data.cc",
+    ],
+    hdrs = [
+        "circular_buffer_flexbuffers_generated_data.h",
+    ],
+)
```
[the remark this answers is not in the document]
Comment (njeffrie) [reply tensorflow/lite/micro/kernels/BUILD]: Done.

File: tensorflow/lite/micro/tools/make/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (njeffrie) [reply tensorflow/lite/micro/tools/make/Makefile:316]: Let's do this in a follow-on to both flexbuffers cc files?

File: tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc (hunk 2 above)
Comment (njeffrie) [reply tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc:69]: Done.

File: tensorflow/lite/micro/kernels/test_data_generation/BUILD [hunk 4]
```diff
@@ -19,3 +19,18 @@ build_test(
...
+    ],
+    deps = [
+        "//third_party/flatbuffers",
+    ],
+)
+
+build_test(
+    name = "build_circular_buffer_generator_test",
+    targets = [":generate_circular_buffer_flexbuffers_data"],
+)
```
[the remark this answers is not in the document]
Comment (njeffrie) [reply tensorflow/lite/micro/kernels/test_data_generation/BUILD:36]: Done.

File: tensorflow/lite/micro/kernels/circular_buffer.cc [hunk 5]
```diff
@@ -96,20 +109,22 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
   // TODO(b/150001379): Move this special case logic to the tflite flatbuffer.
   static int cb_prepare_count = 0;
   cb_prepare_count++;
```
[the remark this answers is not in the document]
Comment (njeffrie) [reply tensorflow/lite/micro/kernels/circular_buffer.cc:111]: Moved.

File: tensorflow/lite/micro/tools/make/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (advaitjain) [reply tensorflow/lite/micro/tools/make/Makefile:316]: sounds good.

**Review by advaitjain — APPROVED** [review]
Looks good!

The force push is preventing me from seeing changes since my last review. Let's see if we can avoid that in the future.

File: tensorflow/lite/micro/tools/make/Makefile [hunk 6]
```diff
@@ -330,7 +330,7 @@ tensorflow/lite/micro/kernels/elu.cc \
 tensorflow/lite/micro/kernels/ethosu.cc \
 tensorflow/lite/micro/kernels/exp.cc \
 tensorflow/lite/micro/kernels/fill.cc \
-tensorflow/lite/micro/kernels/flexbuffers_generated_data.cc \
+tensorflow/lite/micro/kernels/detection_postprocess_flexbuffers_generated_data.cc \
```
Comment (advaitjain) [inline tensorflow/lite/micro/tools/make/Makefile:333]: since we're going with explicitly listing the test targets, we should be able to remove this line.

Comment (njeffrie) [reply tensorflow/lite/micro/tools/make/Makefile]: Done.

**Pull request merged**